src/cmd/sanity_test/sanity_tests.go

```diff
@@ -323,91 +328,120 @@ func checkOnedriveRestoration(
 			fileSizes[itemName] = ptr.Val(driveItem.GetSize())
 		}
 
-		folderPermission[itemID] = permissionsIn(ctx, client, driveID, itemID, folderPermission[itemID])
+		if driveItem.GetFolder() == nil && driveItem.GetPackage() == nil {
+			continue
+		}
+
+		// currently we don't restore blank folders.
+		// skip permission check for empty folders
+		if ptr.Val(driveItem.GetFolder().GetChildCount()) == 0 {
+			fmt.Println("skipped empty folder: ", itemName)
+			continue
+		}
+
+		permissionIn(ctx, client, driveID, itemID, itemName, folderPermission)
+		getOneDriveChildFolder(ctx, client, driveID, itemID, itemName, fileSizes, folderPermission, startTime)
+	}
+
+	getRestoreData(ctx, client, *drive.GetId(), restoreFolderID, restoreFile, restoreFolderPermission, startTime)
+
+	for checkFolderName, checkfolderPer := range folderPermission {
+		fmt.Printf("checking for folder: %s \n", checkFolderName)
+
+		if len(checkfolderPer) < 1 {
+			fmt.Printf("no permissions found for folder : %s.", checkFolderName)
+			continue
+		}
+
+		if len(restoreFolderPermission[checkFolderName]) < 1 {
+			fmt.Printf("permissions are not equal")
+			fmt.Println("Item:", checkFolderName)
+			fmt.Println("Permission found: ", checkfolderPer)
+			fmt.Println("blank permission found in restore.")
+
+			os.Exit(1)
+		}
+
+		for i, orginalFolderPer := range checkfolderPer {
+			if !(orginalFolderPer.entityID != restoreFolderPermission[checkFolderName][i].entityID) &&
+				!slices.Equal(orginalFolderPer.roles, restoreFolderPermission[checkFolderName][i].roles) {
+				fmt.Printf("permissions are not equal")
+				fmt.Printf("*  expected role: %+v \n", orginalFolderPer.roles)
+				fmt.Printf("*  actual:  %+v \n", restoreFolderPermission[checkFolderName][i].roles)
+				fmt.Printf("* entitiy ID expected: %+v \n", orginalFolderPer.entityID)
+				fmt.Printf("* entitiy ID actual:  %+v \n", restoreFolderPermission[checkFolderName][i].entityID)
+				fmt.Println("Item:", checkFolderName)
```

Comment:
This feels a bit strange to print: you didn't compare the IDs for equality, but this print out reads like they're part of the comparison that failed.  I recommend rolling this all into the final metadata line.
```suggestion
				fmt.Printf("* entitiy ID expected: %+v \n", orginalFolderPer.entityID)
				fmt.Printf("* entitiy ID actual:  %+v \n", restoreFolderPermission[checkFolderName][i].entityID)
				fmt.Printf("Item: %s\nOriginal Entity ID: %s\nRestored Entity ID: %s\n", folderName, originalPermission.entityID, restoredPermission.entityID)
```